### PR TITLE
Refactor CreateSprintTemplate for easier unit testing and add basic tests

### PR DESCRIPTION
**Pull request by jayeclark**
## Description
Now that we've got the GitHub actions CI/CD is mostly functioning again (seems like there might still be some issues with the vercel deployment), I'd like to do some work to flesh out the unit & integration tests so that they can actually automatically catch breaking changes in the code. Hopefully with a few example PRs, we can also get new contributors participating as well!

This is the first in a series of PRs to flesh out unit tests related to the arena components (see `arena` folder of repository file structure). Going to leave it open for a bit for review/comments but I don't think this should be too controversial, so will probably merge myself in a few days if no feedback.

* Refactors component to enable external dependencies to more easily be mocked
* Adds some very basic unit tests just verifying that the initial state of the component is as expected
* Additional tests in the future will focus on form validation behavior, column drag-drop behavior, and edge cases (large amounts of data, failed API calls, etc.)

## Relates to
No specific issue, general housekeeping/developer experience improvement

## Reviewers
- @mikhael28 
- @jeremydthomas 

## Screenshots / other info

Screen Shot 2023-02-01 at 6 43 11 PM

### CODE REVIEW

File: src/arena/CreateSprintTemplate.tsx [hunk 1]
```diff
@@ -91,7 +91,7 @@ function CreateSprintTemplate(props: CreateSprintTemplateProps) {
   const { handleShowError, handleShowSuccess } = useContext(ToastMsgContext);
 
   const theme = useTheme();
-  const navigate = useNavigate();
+  const navigate = props.navigate;
```
Comment (jeremydthomas) [inline src/arena/CreateSprintTemplate.tsx:94]: Just a random question, why the change to props.navigate instead of usenavigate hook?

**Review by jeremydthomas — APPROVED** [review]
It all looks good to me, as for the tests Im not strong enough in jest to have an opinion. Cypress is the only one I've used. Oh I guess Ive used mocha and chai a bit.  It LOOKS right to me though. Everything still works as well.

**Review by mikhael28 — APPROVED** [review]

**Comment by jayeclark** [comment]
I'm going to push another commit to try and take care of some of the remaining DeepSource complaints. 😁

File: src/arena/CreateSprintTemplate.tsx (hunk 1 above)
Comment (jayeclark) [reply src/arena/CreateSprintTemplate.tsx]: It's mainly to make unit testing easier. If we pass the navigation function as a prop, then we can easily mock it and prevent it from throwing an error. 

useNavigate can only be used in the context of a Router, so if we render the component for testing _without_ a router context, the test will fail.  Calling useNavigate() will throw an error and prevent the component from rendering. 

Another alternative would be to wrap the rendered component in a generic router component in the test file, ie:

```typescript
// CreateSprintTemplate.spec.tsx

62  render(
63    <BrowserRouter>
64      <CreateSprintTemplate
65       user={testUser}
66        navigate={mockNavigate}
67        getTemplates={mockGetTemplates}
68        setTemplate={mockSetTemplate}
69        getTemplateOptionsFromSanity={mockGetTemplateOptions}
70      />
71    </BrowserRouter>
72  );
```
Definitely open to debate on which approach is better. I personally prefer the simplicity of just rendering the specific component in the test, but the trade-off is a bit more complexity in how the component is set up and what needs to be mocked in the unit tests.

Though, thanks for mentioning it - I took a second look and how I did it needed to be tweaked. It wouldn't cause issues with how the repository is currently set up, but if we get fancier with navigation in the future it could cause issues.

Eventually I'll probably switch to the strategy discussed here: https://blog.logrocket.com/testing-react-router-usenavigate-hook-react-testing-library/ or I might put it up as an issue for someone to work on.

Comment (jeremydthomas) [reply src/arena/CreateSprintTemplate.tsx]: Awesome, I just wanted to know your thinking, because I didnt know any better :)
